marklogic-data-hub-central/ui/src/components/explore/entity-specific-sidebar/entity-specific-sidebar.tsx

```diff
@@ -0,0 +1,364 @@
+  entityFacets: any,
+  checkFacetRender: (facets: any) => void;
+  facetRender: (facets: any) => void;
+  updateSpecificFacets: any;
+}
+
+const EntitySpecificSidebar: React.FC<Props> = (props) => {
+  const {
+    entitySelected: {name, icon},
+    entityFacets,
+    updateSpecificFacets,
+    checkFacetRender,
+    facetRender
+  } = props;
+  // const entityIcon = Icons[icon];
+
+  const {
+    searchOptions,
+    clearConstraint,
+    clearFacet,
+    clearGreyFacet,
+    clearRangeFacet,
+    clearGreyRangeFacet,
+    greyedOptions,
+    setAllGreyedOptions,
+    // setQueryGreyedOptions
+  } = useContext(SearchContext);
+
+  const [allSelectedFacets, setAllSelectedFacets] = useState<any>(searchOptions.selectedFacets);
+  // const [entitySpecificSearch, setEntitySpecificSearch] = useState<string>("");
+
+  let integers = ["int", "integer", "short", "long"];
+  let decimals = ["decimal", "double", "float"];
+
+  useEffect(() => {
+    if (entityFacets || updateSpecificFacets) {
+      if (Object.entries(searchOptions.selectedFacets).length !== 0) {
+        let selectedFacets: any[] = [];
+        for (let constraint in searchOptions.selectedFacets) {
+          let displayName = "";
+          let entityFacet = entityFacets && entityFacets.find(facet => facet.facetName === constraint);
+          if (entityFacet && entityFacet.propertyPath !== constraint) {
+            displayName = entityFacet.propertyPath;
+          }
+          if (constraint === "createdOnRange") {
+            selectedFacets.push({constraint, facet: searchOptions.selectedFacets[constraint], displayName});
+          } else {
+            let datatype = searchOptions.selectedFacets[constraint].dataType;
+            if (datatype === "xs:string" || datatype === "string") {
+              searchOptions.selectedFacets[constraint]["stringValues"].forEach(facet => {
+                selectedFacets.push({constraint, facet, displayName});
+              });
+            } else if (integers.includes(datatype) || decimals.includes(datatype)) {
+              let rangeValues = searchOptions.selectedFacets[constraint].rangeValues;
+              selectedFacets.push({constraint, rangeValues, displayName});
+            } else if (datatype === "xs:date" || datatype === "date") {
+              let rangeValues = searchOptions.selectedFacets[constraint].rangeValues;
+              selectedFacets.push({constraint, rangeValues, displayName});
+            } else if (datatype === "xs:dateTime" || datatype === "dateTime") {
+              let rangeValues = searchOptions.selectedFacets[constraint].rangeValues;
+              selectedFacets.push({constraint, rangeValues, displayName});
+            }
+          }
+          facetRender(selectedFacets);
+        }
+      } else {
+        if (updateSpecificFacets) {
+          facetRender([]);
+          setAllSelectedFacets({});
+        }
+      }
+    }
+  }, [searchOptions.selectedFacets, updateSpecificFacets]);
```

Comment:
I'm not understanding why we ahve this flag updateSpecificFacets

Reply:
@aniamy this property is used for updating the component